var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -195,7 +195,8 @@ class Mfem(Package, CudaPackage, ROCmPackage):
 
     depends_on('mpi', when='+mpi')
     depends_on('hypre@2.10.0:2.13.99', when='@:3.3.99+mpi')
-    depends_on('hypre', when='@3.4:+mpi')
+    depends_on('hypre@:2.20.0', when='@3.4:4.2.0+mpi')
+    depends_on('hypre', when='@4.2.1:+mpi')
```

Comment:
The next MFEM release will be 4.3, so it makes sense to use:
```suggestion
    depends_on('hypre@:2.20.0', when='@3.4:4.2.99+mpi')
    depends_on('hypre', when='@4.3.0:+mpi')
```